src/main/java/org/ambraproject/rhino/view/article/ItemSetView.java

```diff
@@ -45,7 +45,7 @@
     private HibernateTemplate hibernateTemplate;
 
     public ItemSetView getView(ArticleIngestion ingestion) {
-      Collection<ArticleFile> files = (List<ArticleFile>) hibernateTemplate.find("FROM ArticleFile WHERE ingestion = ?", ingestion);
+      Collection<ArticleFile> files = (List<ArticleFile>) hibernateTemplate.find("SELECT file FROM ArticleFile file LEFT JOIN FETCH file.item WHERE file.ingestion = ?", ingestion);
```

Comment:
actually maybe I'm wrong about the `fetch = FetchType.EAGER` above being totally useless. There's a few references to `getFiles()` elsewhere in the code:
https://github.com/PLOS/rhino/blob/174a0db82e0602ddfbf820855aad4cc7229fdea1/src/main/java/org/ambraproject/rhino/view/article/ItemSetView.java#L54
and 
https://github.com/PLOS/rhino/blob/174a0db82e0602ddfbf820855aad4cc7229fdea1/src/main/java/org/ambraproject/rhino/model/ArticleItem.java#L136

is it ok that these no longer eagerly loading `files`?

Reply:
It took me a while to wrap my head around this again.

In the data model, an ingestion has-many items, and an item has-many files. In practice, I think the files are used for derivative images, e.g. thumbnails, so articles typically have many items (> 10 at least) but each item typically has at most a few files. So in our sample article there are over 1000 items.

For some reason the `ItemSetView` class takes an ingestion and a collection of files, not items. It then groups the files by item for display. (https://github.com/plos/rhino/blob/eliminiate-n-plus-one-queries-in-items/src/main/java/org/ambraproject/rhino/view/article/ItemSetView.java#L60)

I played around with, as I recall, various combinations of setting eager load without any luck. It doesn’t help that for some reason the `ArticleIngestion` does not seem to have a `@ManyToOne` relation to `ArticleItem`.

This is the only combination that I could come up with that did not generate n+1 queries. I admit I’m not totally clear what the implications of lazy-loading the files will be, but in theory it should just be an optimization to eager load.

Reply:
so if I understand you correctly, the change on https://github.com/PLOS/rhino/pull/451/files#diff-592dbf86987f3f176ce26932fcb81105L68 is actually necessary to fixing the n+1 query and wasn't merely a semi-related cleanup. Is that right?

Since, as you mention, we're starting with an ingestion and a collection of file objects, It's still unclear to me what altering the association on `ArticleItem` does, since doesn't seem to be used here (rather, the inverse association would be used).

Maybe I can hash this out with you on a call today.